### PR TITLE
Add the support of configuring the default timeout of TaskRun

### LINKED ISSUE #979: The default TaskRun timeout is too short.
# Expected Behavior

A user shouldn't have their `TaskRun`s killed in 10 minutes if they haven't specified a larger `PipelineRun` timeout.

# Actual Behavior

The `TaskRun`s do get killed.

# Steps to Reproduce the Problem

1. Kinda obvious, really. =)

# Additional Info

We should bump this up to something more reasonable. Also relates to #978.

### PR DESCRIPTION
**Pull request by houshengbo**
# Changes

* Changes the default timeout from 10 mins to 60 mins
* Adds a config_defaults configMap for configuring pipeline default values per tektoncd/pipeline instance

Closes #979

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
The default timeout for TaskRuns and PipelineRuns is increased from 10 minutes to 60 minutes. This timeout will now be configurable.
```

### CODE REVIEW

**Comment by bobcatfish** [comment]
I think part of your change might be missing from the commit, looking at https://github.com/tektoncd/pipeline/pull/996/commits/2b10d2436b3af7529270039fa40a91725909133b it says " Add a new param default_timeout for TaskRun" but the changes look like they only do the first part "Change the default timeout from 10 mins to 60 mins".

Looks like a step in the right direction anyway!

(I hope you don't mind @houshengbo I updated the "release note" section of the PR description)

**Comment by bobcatfish** [comment]
> I think part of your change might be missing from the commit, 

Or maybe it's b/c this was a "WIP" in progress review, in which case my apologies for reviewing too early! :pray:

**Comment by houshengbo** [comment]
@bobcatfish Thank you for the updates. 
It is still under development.

**Comment by abayer** [comment]
@houshengbo Looks like codegen needs to be run. =)

**Comment by houshengbo** [comment]
@abayer @bobcatfish @vdemeester 
I need some suggestions from you folks. I created a store to load the configuration of the default timeout as defined in config-defaults.yaml. Where do I need to load the data, by calling NewStore()?

Should I do it in reconciler for pipelinerun and taskrun? Or should I call is somewhere, so that global context will have it, and can be accessed anywhere in need?

**Comment by abayer** [comment]
I'm wondering if this needs to be this elaborate? We use `config-artifact-bucket` and `config-artifact-pvc` just via `c.CoreV1().ConfigMap(ns).Get(name, metav1.GetOptions{})` - is that something that we don't think will scale well as we keep adding more configuration?

**Comment by houshengbo** [comment]
@abayer 
I would rather use an existing one, but
do u think config-artifact-pvc or config-artifact-bucket are good candidates to add a global default timeout?

**Comment by abayer** [comment]
I have no problem with creating a new one, and making it more general, I'm just wondering if we need to be using the approach you're using here to get data from it or if we can use something more like we do with the existing ones.

File: pkg/apis/config/default.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
```
Comment (vdemeester) [inline pkg/apis/config/default.go]: imports needs to be formatted :angel: (`goimports` on save file is :heart: :wink: )

File: pkg/apis/config/default.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	"strconv"
+)
+
+const (
+	// ConfigName is the name of the configmap
+	DefaultsConfigName       = "config-defaults"
+    defaultTimeoutMinutesKey = "default-timeout-minutes"
```
Comment (vdemeester) [inline pkg/apis/config/default.go]: indentation is wrong here

**Comment by houshengbo** [comment]
@dlorenc @ImJasonH Could you help me review this PR? Thx.

**Comment by houshengbo** [comment]
@vdemeester Already did it.

**Comment by bobcatfish** [comment]
Thanks for making this change @houshengbo ! Would it also make sense to add some docs at https://github.com/tektoncd/pipeline/blob/master/docs/install.md#configuring-tekton-pipelines so users know this is something they can configure?

**Pull request merged**